gRPC retry metadata (#27223)
dotnet/AspNetCore.Docs: aspnetcore/grpc/retries.md

The actual delay between retry attempts is randomized. A randomized delay between 0 and the current backoff determines when the next retry attempt is made. Consider that even with exponential backoff configured, increasing the current backoff between attempts, the actual delay between attempts isn't always larger. The delay is randomized to prevent retries from multiple calls from clustering together and potentially overloading the server.

> ### Detect retried calls with metadata

Author: ```suggestion
### Detect retries with metadata
```

> Retried gRPC calls can be detected by the `grpc-previous-rpc-attempts` metadata. The metadata value is:

Author: ```suggestion
gRPC retries can be detected by the `grpc-previous-rpc-attempts` metadata. The metadata value is:
```

> gRPC retries can be detected by the presence of `grpc-previous-rpc-attempts` metadata. The `grpc-previous-rpc-attempts` metadata is:

Reviewer: ```suggestion
gRPC retries can be detected by the presence of `grpc-previous-rpc-attempts` metadata. The `grpc-previous-rpc-attempts` metadata:
```

> gRPC retries can be detected by the `grpc-previous-rpc-attempts` metadata. The `grpc-previous-rpc-attempts` metadata is:
> 
> * Added by the client to retried calls and sent to the server.

Author: ```suggestion
gRPC retries can be detected by the presence of `grpc-previous-rpc-attempts` metadata. The `grpc-previous-rpc-attempts` metadata is:

* Automatically added to retried calls and sent to the server.
```

> * Automatically added to retried calls and sent to the server.

Reviewer: ```suggestion
* Is automatically added to retried calls and sent to the server.
```

Reviewer: This seems to work a bit smoother for all three statements in the list.

* Value represents the number of preceding retry attempts.
* Value is always an integer.

Consider the following retry scenario:

1. Client makes a gRPC call to the server.
2. Server fails and returns a retriable status code response.

> 3. Client retries the gRPC call. Because there was one previous attempt, `grpc-previous-rpc-attempts` metadata has a value of `1`. It is sent to the server with the retry.

Reviewer: ```suggestion
3. Client retries the gRPC call. Because there was one previous attempt, `grpc-previous-rpc-attempts` metadata has a value of `1`. The `grpc-previous-rpc-attempts` metadata is sent to the server with the retry.
```

Reviewer: "It" leaves a bit of room for misunderstanding what was sent.

4. Server succeeds and returns OK.
5. Client reports success. `grpc-previous-rpc-attempts` is in the response metadata and has a value of `1`.

> The metadata is not present on the initial gRPC call, is `1` for the first retry, `2` for the second retry, and so on.

Author: ```suggestion
The `grpc-previous-rpc-attempts` metadata is not present on the initial gRPC call, is `1` for the first retry, `2` for the second retry, and so on.
```
